### PR TITLE
DBAL Transport

### PR DESCRIPTION
**Pull request by ASKozienko**

### CODE REVIEW

File: pkg/dbal/DbalConnectionFactory.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        return new DbalContext($this->establishConnection(), $this->config);
+    }
+
+    /**
+     * @return Connection
+     */
+    private function establishConnection()
+    {
+        return $this->registry->getConnection($this->config['connectionName']);
```
Comment (makasim) [inline pkg/dbal/DbalConnectionFactory.php]: I guess we should call connect at this stage

File: pkg/dbal/DbalConsumer.php [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+                $this->dbal->commit();
+
+                return $message;
+            }
+
+            $this->dbal->commit();
+        } catch (\LogicException $e) {
+            $this->dbal->rollBack();
+            throw $e;
+        } catch (\Exception $e) {
```
Comment (makasim) [inline pkg/dbal/DbalConsumer.php]: why do you need two catch statements  here?

File: pkg/dbal/DbalConsumer.php [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+     * @return DbalMessage
+     */
+    protected function convertMessage(array $dbalMessage)
+    {
+        $message = $this->context->createMessage();
+
+        $message->setId($dbalMessage['id']);
+        $message->setBody($dbalMessage['body']);
+        $message->setPriority((int) $dbalMessage['priority']);
+        $message->setRedelivered((bool) $dbalMessage['redelivered']);
```
Comment (makasim) [inline pkg/dbal/DbalConsumer.php]: there is no such thing for now

File: pkg/dbal/Symfony/DbalTransportFactory.php [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addConfiguration(ArrayNodeDefinition $builder)
+    {
+        $builder
+            ->children()
+                ->scalarNode('connectionName')
```
Comment (bendavies) [inline pkg/dbal/Symfony/DbalTransportFactory.php]: other transport factories use snake_case

Comment (makasim) [inline pkg/dbal/Symfony/DbalTransportFactory.php]: +1

**Comment by makasim** [comment]
partly address https://github.com/php-enqueue/enqueue-dev/issues/32

File: pkg/dbal/Client/DbalDriver.php [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+        $clientMessage->setBody($message->getBody());
+        $clientMessage->setHeaders($message->getHeaders());
+        $clientMessage->setProperties($message->getProperties());
+
+        $clientMessage->setContentType($message->getHeader('content_type'));
+        $clientMessage->setMessageId($message->getMessageId());
+        $clientMessage->setTimestamp($message->getTimestamp());
+        $clientMessage->setPriority(MessagePriority::NORMAL);
+        $clientMessage->setDelay($message->getDelay());
```
Comment (makasim) [inline pkg/dbal/Client/DbalDriver.php]: setReplyTo
setCorreltationId

File: pkg/dbal/Client/DbalDriver.php [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+        $headers = $message->getHeaders();
+        $headers['content_type'] = $message->getContentType();
+
+        $transportMessage = $this->context->createMessage();
+        $transportMessage->setBody($message->getBody());
+        $transportMessage->setHeaders($headers);
+        $transportMessage->setProperties($properties);
+        $transportMessage->setMessageId($message->getMessageId());
+        $transportMessage->setTimestamp($message->getTimestamp());
+        $transportMessage->setDelay($message->getDelay());
```
Comment (makasim) [inline pkg/dbal/Client/DbalDriver.php]: replyTo,correlationId

File: pkg/dbal/Client/DbalDriver.php [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    public function sendToRouter(Message $message)
+    {
+        if (false == $message->getProperty(Config::PARAMETER_TOPIC_NAME)) {
+            throw new \LogicException('Topic name parameter is required but is not set');
+        }
+
+        $topic = $this->createRouterTopic();
```
Comment (makasim) [inline pkg/dbal/Client/DbalDriver.php]: better to use 

$queue = $this->createQueue($this->config->getRouterQueueName());

File: pkg/dbal/DbalConsumer.php [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+    {
+        return $this->queue;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @return DbalMessage|null
+     */
+    public function receive($timeout = 0)
```
Comment (makasim) [inline pkg/dbal/DbalConsumer.php]: timeout in miliseconds, if you need seconds do `$timeout / 1000`

File: pkg/dbal/LICENSE [hunk 9]
```diff
@@ -0,0 +1,20 @@
+The MIT License (MIT)
+Copyright (c) 2016 Kotliar Maksym
```
Comment (makasim) [inline pkg/dbal/LICENSE]: oro copyright

File: pkg/dbal/Tests/Client/DbalDriverTest.php [hunk 10]
```diff
@@ -0,0 +1,318 @@
...
+    public function testShouldImplementsDriverInterface()
+    {
+        $this->assertClassImplements(DriverInterface::class, DbalDriver::class);
+    }
+
+    public function testCouldBeConstructedWithRequiredArguments()
+    {
+        new DbalDriver(
+            $this->createPsrContextMock(),
+            new Config('', '', '', '', '', '')
```
Comment (makasim) [inline pkg/dbal/Tests/Client/DbalDriverTest.php]: Config::create()

File: pkg/dbal/composer.json [hunk 11]
```diff
@@ -0,0 +1,41 @@
+{
+    "name": "enqueue/dbal",
+    "type": "library",
+    "description": "Message Queue Doctrine DBAL Transport",
+    "keywords": ["messaging", "queue", "dbal"],
```
Comment (makasim) [inline pkg/dbal/composer.json]: doctrine

**Comment by makasim** [comment]
LGTM, you can start with SQS. I'll take care of the rest.

**Pull request merged**